pkg/skaffold/kubernetes/status/status_check.go

```diff
@@ -183,17 +191,24 @@ func (s *Monitor) statusCheck(ctx context.Context, out io.Writer) (proto.StatusC
 	// Wait for all deployment statuses to be fetched
 	wg.Wait()
 	cancel()
+	// update seen all pods retrieved.
+	s.updateSeenResources(deployments)
 	return getSkaffoldDeployStatus(c, deployments)
 }
 
-func getDeployments(ctx context.Context, client kubernetes.Interface, ns string, l *label.DefaultLabeller, deadlineDuration time.Duration) ([]*resource.Deployment, error) {
+func getDeployments(ctx context.Context, client kubernetes.Interface, ns string, l *label.DefaultLabeller, deadlineDuration time.Duration, prevPods resource.Group) ([]*resource.Deployment, error) {
 	deps, err := client.AppsV1().Deployments(ns).List(ctx, metav1.ListOptions{
 		LabelSelector: l.RunIDSelector(),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("could not fetch deployments: %w", err)
 	}
 
+	var ignore []string
+	for k := range prevPods {
```

Comment:
nit: Are `prevPods` only the pods that did not come up successfully the prev iteration?  Would `unhealthyPods` possibly be more descriptive/accurate here?  I would think only unhealthy pods are added to the ignore list.

Reply:
From this I'm a bit confused, is the idea to ignore all pods from prev iteration or only the unhealthy pods from the prev iteration (as per the title?)  Are they the same thing?  (this wouldn't be obvious to me as I would think the prev itertion some might have been healthy, is the idea to not report any logs until all are healthy?)

Reply:
Previous pods would just be any pod from the previous iteration. This change is mostly so we ignore them in the event API. CLI users shouldn't see any difference as we don't print updates from pods normally anyway, however we do send out events for them. So this will help IDE's not report pod failures when we tear down the old pods.

Reply:
nope. They could be unhealthy or healthy pods.  
use case we have in mind is [this](https://github.com/GoogleCloudPlatform/cloud-code-intellij-internal/issues/4323#issue-951100782)

Here if you see as we are tearing down previous iteration due to `kubectl apply` event (healthy and unhealthy) related to those pods (since they have the same run-id) are shown.